api.wsgi

```diff
@@ -24,85 +28,96 @@ ap.add_argument('--api_uri', help='api uri, with https:// prefix')
 ap.add_argument('--insecure', help='allow user info as urlencoded param', action='store_true', default=False)
 ap.add_argument('--central_uri', help='scitran central api', default='https://sdmc.scitran.io/api')
 ap.add_argument('--log_level', help='log level [info]', default='info')
+ap.add_argument('--drone_secret', help='shared drone secret')
+
+if __name__ == '__main__':
+    import paste.httpserver
+    logging.getLogger('paste.httpserver').setLevel(logging.WARNING) # silence paste logging
+
+    ap.add_argument('--host', default='127.0.0.1', help='IP address to bind to [127.0.0.1]')
+    ap.add_argument('--port', default='8080', help='TCP port to listen on [8080]')
+
 args = ap.parse_args()
+log.setLevel(getattr(logging, args.log_level.upper()))
 
-# HACK to allow setting the --site_name in the same way as api.py
-# --site_name 'Example Site' or --site_name "Example Site"
+# uwsgi-related HACK to allow --site_name 'Example Site' or --site_name "Example Site"
 args.site_name = ' '.join(args.site_name).strip('"\'')
+
 args.quarantine_path = os.path.join(args.data_path, 'quarantine')
 args.upload_path = os.path.join(args.data_path, 'upload')
 
-logging.basicConfig(level=getattr(logging, args.log_level.upper())) #FIXME probably not necessary, because done in api.py
-log = logging.getLogger('scitran')
-
-# configure uwsgi application
-application = api.app
-application.config = vars(args)
-
-if not os.path.exists(application.config['data_path']):
-    os.makedirs(application.config['data_path'])
-if not os.path.exists(application.config['quarantine_path']):
-    os.makedirs(application.config['quarantine_path'])
-if not os.path.exists(application.config['upload_path']):
-    os.makedirs(application.config['upload_path'])
-if not application.config['apps_path']:
-    log.warning('apps_path is not defined.  Apps functionality disabled')
-else:
-    if not os.path.exists(application.config['apps_path']):
-        os.makedirs(application.config['apps_path'])
+api.app.config = vars(args)
+
+centralclient_enabled = True
+if not api.app.config['ssl_cert']:
+    centralclient_enabled = False
+    log.warning('ssl_cert not configured -> SciTran Central functionality disabled')
+if api.app.config['site_id'] == 'local':
+    centralclient_enabled = False
+    log.warning('site_id not configured -> SciTran Central functionality disabled')
+elif not api.app.config['api_uri']:
+    centralclient_enabled = False
+    log.warning('api_uri not configured -> SciTran Central functionality disabled')
+if not api.app.config['central_uri']:
+    centralclient_enabled = False
+    log.warning('central_uri not configured -> SciTran Central functionality disabled')
+if not api.app.config['drone_secret']:
+    log.warning('drone_secret not configured -> Drone functionality disabled')
+if not api.app.config['apps_path']:
+    log.warning('apps_path is not defined -> App functionality disabled')
```

Comment:
Hmm, this set of checks makes me want a json schema for the loaded config map :)

Maybe we could define one and add it to `check-version.py` in the infra.
If that sounds like a good idea I'll add a ticket over there about it.